tests/test_writer.py

```diff
@@ -55,3 +55,16 @@ def test_format_requirement_not_for_primary(from_line, writer):
                                        reverse_dependencies,
                                        primary_packages=['test']) ==
             'test==1.2')
+
+
+def test_format_requirement_environment_marker(from_line, writer):
+    "Primary packages should not get annotated."
+    ireq = from_line("test ; python_version == '2.7' and platform_python_implementation == 'CPython'")
+    reverse_dependencies = set()
+
+    result = writer._format_requirement(
+        ireq, reverse_dependencies, primary_packages=['test'],
+        marker=ireq.markers)
+    result = result.replace('\"', '\'')
```

Comment:
I was seeing different behavior in Python 2.7 vs 3.5. Python 2.7 preserved quotes in the original format, while Python 3.5 was converting to double quotes. I assume this is a difference in the underlying pip library which parses the lines (including environment markers) from `requirements.in`.